tests/protocols/analytics_test.py

```diff
@@ -0,0 +1,58 @@
+from client.protocols import analytics
+from os import path
+
+import mock
+import unittest
+
+class TestAnalyticsProtocol(unittest.TestCase):
+    def setUp(self):
+        self.cmd_args = mock.Mock()
+        self.cmd_args.submit = True
+        self.assignment = mock.MagicMock()
+        self.file_dir = path.abspath(path.join(path.dirname(__file__), '..', "data"))
+
+        self.proto = analytics.protocol(self.cmd_args, self.assignment)
+
+    def read_file(self, file_name):
+        file_path = path.join(self.file_dir, file_name)
+        print(file_path)
+        with open(file_path, 'r', encoding="utf-8") as lines:
+            return lines.read()
+
+    def call_check_start(self, files):
+        q_status = self.proto.check_start(files)
+
+        self.assertNotEqual(q_status, {})
+        return q_status
+
+    def testOnCheckStart_questionStarted(self):
+        files = {'test1': self.read_file("questionStarted.py")}
+        self.assertEqual({
+            'Question 1': True,
```

Comment:
@albert12132 Do you mean the string "Question 1" here? If so, they are actually read from the test "files" and they might change based on the tag contents (e.g. the tag could be "BEGIN Q1" instead of "BEGIN Question 1" and the answer should contain "Q1" as the key), so I'm not sure if it's necessary to make them as class variables.